pipeline/src/executor.rs

```diff
@@ -26,30 +23,48 @@ impl Executor {
             .stdout(Stdio::piped())
             .stderr(Stdio::piped())
             .spawn()
-            .unwrap();
+            .expect("Couldn't spawn child process");
 
-        let stdout = BufReader::new(child.stdout.take().unwrap());
-        let stderr = BufReader::new(child.stderr.take().unwrap());
+        let stdout = BufReader::new(child.stdout.take().expect("Couldn't take stdout of child"));
+        let stderr = BufReader::new(child.stderr.take().expect("Couldn't take stderr of child"));
 
         let stdout_handle = thread::spawn(move || {
-            stdout.lines().for_each(|line| {
-                println!("[docker] {}", line.unwrap());
-            });
-        });
-        stderr.lines().for_each(|line| {
-            let line = line.unwrap();
-            println!("[zkvm-stdout] {}", line);
-            if self.test && line.contains("Executable unittests") {
-                if let Some(test) = extract_path(&line) {
-                    self.artifacts
-                        .iter_mut()
-                        .find(|a| file_name(&test).contains(&file_name(a).replace('-', "_")))
-                        .map(|a| *a = test)
-                        .expect("Failed to find test artifact");
-                }
+            for line in stdout.lines().enumerate().map(|(index, line)| {
+                line.unwrap_or_else(|e| {
+                    panic!("Couldn't get stdout line: {index}\n with error: {e}")
+                })
+            }) {
+                println!("[docker] {line}");
             }
         });
-        stdout_handle.join().unwrap();
+
+        for line in stderr.lines().enumerate().map(|(index, line)| {
+            line.unwrap_or_else(|e| panic!("Couldn't get stderr line: {index}\n with error: {e}"))
+        }) {
+            println!("[zkvm-stdout] {line}");
+
+            if !self.test || !line.contains("Executable unittests") {
+                continue;
+            }
+
+            let Some(test) = extract_path(&line) else {
+                continue;
+            };
```

Comment:
I kind of prefer the version with the if statements instead of these continues which for me complicate the control flow a bit (early outs).

Reply:
Does this new version look better https://github.com/taikoxyz/raiko/blob/pipeline-refactor/pipeline/src/executor.rs#L46-L58

Reply:
yep!